src/utils/daos_hdlr.c

```diff
@@ -655,6 +656,77 @@ pool_query_hdlr(struct cmd_args_s *ap)
 	return rc;
 }
 
+int
+cont_check_hdlr(struct cmd_args_s *ap)
+{
+	daos_obj_id_t		oids[OID_ARR_SIZE];
+	daos_handle_t		oit;
+	daos_anchor_t		anchor = { 0 };
+	uint32_t		oids_nr;
+	int			rc, i;
+
+	/* Create a snapshot with OIT */
+	rc = daos_cont_create_snap_opt(ap->cont, &ap->epc, NULL,
+				       DAOS_SNAP_OPT_CR | DAOS_SNAP_OPT_OIT,
+				       NULL);
+	if (rc != 0)
+		goto out;
+
+	/* Open OIT */
+	rc = daos_oit_open(ap->cont, ap->epc, &oit, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "open of container's OIT failed: "DF_RC"\n",
+			DP_RC(rc));
+		goto out_snap;
+	}
+
+	while (!daos_anchor_is_eof(&anchor)) {
+		oids_nr = OID_ARR_SIZE;
+		rc = daos_oit_list(oit, oids, &oids_nr, &anchor, NULL);
+		if (rc != 0) {
+			fprintf(stderr,
+				"object IDs enumeration failed: "DF_RC"\n",
+				DP_RC(rc));
+			D_GOTO(out_close, rc);
+		}
+
+		for (i = 0; i < oids_nr; i++) {
+			rc = daos_obj_verify(ap->cont, oids[i], ap->epc);
+			if (rc == -DER_NOSYS) {
+				/* XXX: NOT support to verif EC object yet. */
+				D_PRINT("spik check for object: "DF_OID"\n",
+					DP_OID(oids[i]));
+				rc = 0;
+				continue;
+			}
+
+			if (rc == -DER_MISMATCH) {
+				fprintf(stderr,
+					"found data inconsistency for object: "
+					DF_OID"\n", DP_OID(oids[i]));
+				rc = 0;
+				continue;
+			}
+
+			if (rc < 0) {
+				fprintf(stderr,
+					"check object "DF_OID" failed: "
+					DF_RC"\n", DP_OID(oids[i]), DP_RC(rc));
+				D_GOTO(out_close, rc);
+			}
+
+			D_PRINT("checked object: "DF_OID"\n", DP_OID(oids[i]));
```

Comment:
do we want to print this for every object? we could have millions..

Reply:
using "D_INFO" probably is fine.

Reply:
Maybe we only need to log the inconsistent or skipped objects?

Reply:
I agree with Mohamad. Let's just print a summary at the end.